common/domain/handler_test.go

```diff
@@ -939,3 +939,350 @@ func TestHandler_UpdateIsolationGroups(t *testing.T) {
 		})
 	}
 }
+
+func TestHandler_UpdateAsyncWorkflowConfiguraton(t *testing.T) {
+
+	testDomain := "testDomain"
+
+	tests := []struct {
+		name             string
+		setupMocks       func(mockDomainManager *persistence.MockDomainManager, mockReplicator *MockReplicator)
+		isPrimaryCluster bool
+		request          *types.UpdateDomainAsyncWorkflowConfiguratonRequest
+		expectedErr      error
+	}{
+		{
+			name: "fail to get metadata",
```

Comment:
not a real problem, but just a tip: Genuinely starting with the success case is a bit easier because you typically have to construct the *most* mocks and return results. And thereafter all the error cases are typically just subsets of the first entry. It just makes copying + pasting easier